bpf/lib/srv6.h

```diff
@@ -121,6 +121,54 @@ is_srv6_packet(const struct ipv6hdr *ip6)
 }
 
 # ifndef SKIP_SRV6_HANDLING
+
+#ifdef ENABLE_SRV6_FLOW_LABEL_PORT
+static __always_inline __u16
+calculate_flow_label(__be16 sport, __be16 dport)
+{
+	return (sport + dport) & 0xffff;
+}
```

Comment:
Could you explain a bit about why you chose to use port numbers as a flow label source? I think in general, it is either 5-tuples or 2-tuples (IP addresses).

e.g. https://datatracker.ietf.org/doc/html/rfc6437#section-3

Reply:
Also, Linux's SRv6 implementation uses `skb->hash` which is also accessible from eBPF program (in our case we can get it with `get_hash(ctx)`. Could you consider using it? So that we don't need to parse the inner packet.

https://github.com/torvalds/linux/blob/master/net/ipv6/seg6_iptunnel.c#L109-L125

Reply:
Ack will do in the next commit

Reply:
I changed the code in my lab (not committed yet) and unless I am making some mistake, get_hash(ctx) returns 0x0 value. Could you confirm that it should be returning non-zero value?

Btw. not sure how to interpret the comment in the skb.h source code 

```
/* Avoid expensive calls into the kernel flow dissector if it's not an L4
 * hash. We currently only use the hash for debugging. If needed later, we
 * can map it to BPF_FUNC(get_hash_recalc) to get the L4 hash.
 */
#define get_hash(ctx)		ctx->hash
```

Reply:
Ah, my bad. Yep, we need to call [bpf_get_hash_recalc](https://docs.ebpf.io/linux/helper-function/bpf_get_hash_recalc/) helper to populate it for all cases. Let's make get_hash_recalc macro to be replaced to the actual bpf_get_hash_recalc here.

Reply:
It calls skb_get_hash() internally which is also used for the SRv6 Linux impl.

https://github.com/torvalds/linux/blob/ae90f6a6170d7a7a1aa4fddf664fbd093e3023bc/net/core/filter.c#L3132

Reply:
Please replace all other places that use get_hash_recalc to use get_hash instead. This aligns with what they are doing today and avoids calling an expensive hash recalculation for the existing usecases.

Reply:
My confidence level that the way I have used bpf_get_hash_recalc is somewhat low. The latest change in commit. Hopefully I did it in the right way.

the function get_hash_recalc is used in bpf\lib\common.h 

```
#define __notify_common_hdr(t, s)	\
	.type		= (t),		\
	.subtype	= (s),		\
	.source		= EVENT_SOURCE,	\
	.hash		= get_hash_recalc(ctx)
```

is this where I should be changing to get_has(ctx)?

As well as bpf\include\bpf\ctx\xdp.h. however here, I guess the logic of hash calculation is somewhat different compared to tc level.

```
#define get_hash(ctx)			({ 0; })
#define get_hash_recalc(ctx)		get_hash(ctx)
```

Reply:
> > is this where I should be changing to get_has(ctx)?
> 
> Yes, right. I also searched the code, but seems like this is the only place we're using the `get_hash_recalc`.

commit updated with change